compiler_opt/es/blackbox_learner_test.py

```diff
@@ -0,0 +1,86 @@
+# coding=utf-8
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for blackbox_learner"""
+
+import tempfile
+from absl.testing import absltest
+import concurrent.futures
+
+from compiler_opt.distributed.local import local_worker_manager
+from compiler_opt.es import blackbox_learner, blackbox_optimizers, es_worker
+from compiler_opt.rl import corpus
+
+
+class BlackboxLearnerTests(absltest.TestCase):
+  """Tests for blackbox_learner"""
+
+  learner_config = blackbox_learner.BlackboxLearnerConfig(
+      total_steps=1,
+      blackbox_optimizer='',
+      est_type=blackbox_optimizers.EstimatorType.ANTITHETIC,
+      fvalues_normalization=True,
+      hyperparameters_update_method=blackbox_optimizers.UpdateMethod.NO_METHOD,
+      num_top_directions=0,
+      num_ir_repeats_within_worker=1,
+      num_ir_repeats_across_worker=0,
+      num_exact_evals=1,
+      total_num_perturbations=3,
+      precision_parameter=1,
+      step_size=1.0)
+  cps = corpus.create_corpus_for_testing(
+      location=tempfile.gettempdir(),
+      elements=[corpus.ModuleSpec(name='smth', size=1)],
+      additional_flags=(),
+      delete_flags=())
+  learner = blackbox_learner.BlackboxLearner(
```

Comment:
shouldn't at least some of these go into a setUp() method to ensure they stay unmodified across tests?

Reply:
write:

```
def setUp(self):
  super().setUp()
  self._cps = ...
  self._learner = ...
```

this will be executed as setup prior to every test run.